### PR TITLE
Use shared one connection DB

### PR DESCRIPTION
**Pull request by Denislite**

### CODE REVIEW

**Review by Kaborda-Irina-ScienceSoft — APPROVED** [review]

**Review by alavrovinfb — APPROVED** [review]
LGTM

File: internal/repositories/app.go [hunk 1]
```diff
@@ -41,3 +36,66 @@ func (ar AppRepository) IsExistDeploymentNameInDB(deploymentName string) (bool,
...
+	var allHashDataFromDB []*models.HashDataFromDB
+
+	query := "SELECT id,file_name,full_file_path,hash_sum,algorithm,image_tag,name_pod,name_deployment FROM hashfiles WHERE full_file_path LIKE $1 and algorithm=$2 and name_pod=$3"
+
+	rows, err := ar.db.Query(query, "%"+dirFiles+"%", algorithm, deploymentData.NamePod)
+	if err != nil {
+		ar.logger.Error(err)
+		return nil, err
+	}
+	for rows.Next() {
```
Comment (siarhei-shliayonkin) [inline internal/repositories/app.go:79]: defer rows.Close() - ?

File: internal/repositories/app.go [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
-		return false, err
-	}
-	defer db.Close()
 
 	var count int
 	query := "SELECT COUNT(*) FROM hashfiles WHERE name_deployment=$1 LIMIT 1;"
-	row := db.QueryRow(query, deploymentName)
-	err = row.Scan(&count)
+	row := ar.db.QueryRow(query, deploymentName)
+	err := row.Scan(&count)
```
Comment (siarhei-shliayonkin) [inline internal/repositories/app.go]: TODO: simplify - return true on condition

File: internal/repositories/app.go (hunk 1 above)
Comment (Denislite) [reply internal/repositories/app.go:79]: Done

File: internal/repositories/app.go (hunk 2 above)
Comment (Denislite) [reply internal/repositories/app.go]: Done

Comment (siarhei-shliayonkin) [reply internal/repositories/app.go]: Not exactly.. You've done almost the same, but i mean something like this:
```go
func RowExists(condition string) (rez bool) {
  // general, for any search:
	query := fmt.Sprintf("SELECT TRUE FROM hashfiles WHERE %s", condition)
	_ = db.QueryRow(query).Scan(&rez)
	return // returns true only if found rows that match condition

// in our case we may pass just a name to search, and refactor this to func iSDeploymentExists(name string)
}
...
if RowExists(name_deployment='any_name') {...
```
It's not really important, so leaving it up to you

**Review by siarhei-shliayonkin — APPROVED** [review]

**Pull request merged**